Update AD/LDAP user import routines to work with Authenticator

YaroslavTir opened the pull request:
Fixes https://github.com/OHDSI/ArachneCentralAPI/issues/620
- add UserService class with findUser, and findAllUser methods

Commit:
Update AD/LDAP user import routines to work with Authenticator
- add UserService class with findUser, and findAllUser methods

Commit:
Update AD/LDAP user import routines to work with Authenticator
- wrap "Incorrect result size: expected 1, actual 0" exception

pavgra commented on pom.xml:
```diff
@@ -6,7 +6,7 @@
     <groupId>org.ohdsi</groupId>
     <artifactId>authenticator</artifactId>
-    <version>0.0.2-SNAPSHOT</version>
+    <version>0.0.3-SNAPSHOT</version>
```
The PR is not valid since it introduces breaking changes but only patch version is raised. Overall, I believe that the required functionality could be implemented w/o the breaking changes so that all dependant projects wouldn't be affected

YaroslavTir replied:
fixed

Commit:
Update AD/LDAP user import routines to work with Authenticator. fix PR remarks
- use username, token, authMethod instead of AuthenticationInfo

Commit:
Update AD/LDAP user import routines to work with Authenticator. fix PR remarks
- don't use User object inside of userInfo

Commit:
Update AD/LDAP user import routines to work with Authenticator. refactoring
- use TokenInfo instead of AuthenticationToken
- extract all conversation logic in separate converter classes
- use `fieldsToExtract` properties to populate UserInfo.User object

Commit:
Update AD/LDAP user import routines to work with Authenticator. refactoring
- add getMethod method to authService

Commit:
Update AD/LDAP user import routines to work with Authenticator.
- add getMethodType for AuthServices
- check bad credentials for LDAP connection

Commit:
Update AD/LDAP user import routines to work with Authenticator.
- small clean up of the code

pavgra commented on README.md:
```diff
@@ -47,9 +47,9 @@ The `STANDARD` authentication mode is the default mode, so it is not necessary t
 
 - `fieldsToExtract`:
-  - keys represent names of user's properties, which will be stored in JWT body / UserInfo.additionalInfo
+  - keys represent names of user's properties, which will be stored in JWT body / UserInfo.user
```
Breaking change. Would it be possible to make it back-compatible? i.e. at least preserve `additionalInfo` name

pavgra commented:
After upgrade to the newer version existing tests fail:
![image](https://user-images.githubusercontent.com/10010071/72625243-6bde0e80-3916-11ea-9d38-383b7428f64c.png)

Commit:
Update AD/LDAP user import routines to work with Authenticator.
- make single user query parameterized

pavgra reviewed (approved):
Tested with Terminologist. The single migration requirement was related to the property name change 
 `additionalInfo` -> `user`
